pkg/pillar/cmd/nim/nim.go

```diff
@@ -70,10 +70,11 @@ type nim struct {
 	version       bool
 
 	// NIM components
-	connTester     *conntester.ZedcloudConnectivityTester
-	dpcManager     *dpcmanager.DpcManager
-	dpcReconciler  dpcreconciler.DpcReconciler
-	networkMonitor netmonitor.NetworkMonitor
+	connTester       *conntester.ZedcloudConnectivityTester
+	dpcManager       *dpcmanager.DpcManager
+	initiallyMissing bool // dpcList initially missing
```

Comment:
Even though the associated comment explains it, I would still prefer attribute name that more clearly describes its purpose. Maybe something like `hadDpclOnBoot` or `dpclWasPersisted`.
Also with the refactor, I tried to have these attributes thematically grouped and organized to keep it a bit more readable. [This place](https://github.com/lf-edge/eve/blob/master/pkg/pillar/cmd/nim/nim.go#L73-L76) was meant for NIM components. For configuration and config-related flags, [this](https://github.com/lf-edge/eve/blob/master/pkg/pillar/cmd/nim/nim.go#L104-L109) might be a more suitable place.

Reply:
@milan-zededa Structurally I don't like what I put in the PR. The dpclMissingOnBoot should really be a property of the dpcManager which it can return. But returning from the constructor is odd, and dpcManager doesn't have any other methods which return info. Adding such a function would raise questions whether we need locking inside the dpcManager to be able to read an attribute from its internal struct.

Reply:
@milan-zededa I've changed the name and polarity of the boolean to dpclPresent. dpclPresentOnBoot seems a bit long but I can switch to that if you want. 

But let me know if you think an accessor function is better than having the initializer return this.

Reply:
I think it might make sense to split the DPC Manager initialization and starting of Go routines into separate methods, say `Init() error` and `Run(context) error`.
I have something similar for nim itself: https://github.com/lf-edge/eve/blob/master/pkg/pillar/cmd/nim/nim.go#L119-L124 (although here it is really just for aesthetics)
For DPC Manager, Init would be [this](https://github.com/lf-edge/eve/blob/master/pkg/pillar/dpcmanager/dpcmanager.go#L199-L218) + ingestion of persisted DPCL. Then in the comment of an accessor method (for `dpclPresent`) you would mention that it should be called after Init(). Since it cannot change afterwards, we do not need to lock.
Run method would contain everything related to starting Go routines: https://github.com/lf-edge/eve/blob/master/pkg/pillar/dpcmanager/dpcmanager.go#L220-L227
And inside nim, the calls to Init (and DpclPresent()) and Run of the DPC manager could be split between [nim's init and run methods](https://github.com/lf-edge/eve/blob/master/pkg/pillar/cmd/nim/nim.go#L119-L124).
Regarding naming - `dpclPresent` might invoke that we are checking if DPCL is present in any particular moment (i.e. check if the list is currently not empty). If we want to generalize and have this semantics for the accessor, then we will need to introduce locking for DPC list and things will get slightly more complicated (currently it is accessed only by the manager's main Go routine).